---
title: "Let's Make a Poster"
author: "Jennifer Collister"
date: "13/12/2019"
output: html_document


references:
- id: folley2019
  title: Physical Activity, APOE Genotype, and Cognitive Decline - Exploring Gene-Environment Interactions in the UK Biobank
  author:
  - family: Folley
    given: Stephanie
- id: lyall2016
  title: Alzheimer disease genetic risk factor APOE e4 and cognitive abilities in 111,739 UK Biobank participants
  author:
  - family: Lyall
    given: Donald
- id: lyall2019
  title: Assessing for interaction between APOE e4, sex, and lifestyle on cognitive abilities
  author:
  - family: Lyall
    given: Donald
- id: lyall2016_2
  title: Cognitive test scores in UK Biobank - Data reduction in 480,416 participants and longitudinal stability in 20,346 participants
  author:
  - family: Lyall
    given: Donald


nocite: | 
  @folley2019, @lyall2016, @lyall2019, @lyall2016_2

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(survival)
library(survminer)
library(lmtest)
library(kableExtra)
library(knitr)
library(timereg)
library(yaml)
library(here)

# Specify the markdown format for knitr tables, otherwise they're not compatible with kableExtra
options(knitr.kable.NA='', knitr.table.format = "markdown")
options(kableExtra.auto_format = FALSE)
# Set the root directory to the project directory (otherwise knitr works in the directory of the Rmd file by default)
knitr::opts_knit$set(root.dir = here::here())

# Load the project config file for filepaths etc
config = yaml.load_file(here::here("./config.yml"))

source("K:/TEU/APOE on Dementia/Statistical_Analysis/JCfunctions.R")

apoe <- readRDS(file=paste0(config$analysisdata$apoe, "apoe_surv.rds"))

apoe$e4carrier <- apoe$apoe4>0
apoe$apoe4 <- factor(apoe$apoe4)

apoe$alcdaily <- apoe$Alc_Freq=="Daily or almost daily"
apoe$diab <- apoe$diabetes=="Yes"

# Restrict to White individuals
apoe$cauc <- apoe$eth_group=="White"
apoe <- apoe[apoe$cauc==TRUE,]
# 487310 - 459240 = 28,070 excluded
```

```{r exclusions, include=FALSE}
# Read in the dataset
# n = 514,776
apoe <- readRDS("K:\\TEU\\APOE on Dementia\\Data Management\\R_Dataframes_TLA\\38358\\Organised\\apoe_surv.rds")

## Genetic exclusions

# Exclude individuals with no genetic data
# n= 15,210
apoe <- apoe[!is.na(apoe$apoe_gen),]

# Exclude individuals with APOE e1 because it's rare and don't know much about it
# Exclude individuals with genotype e2/e4 because they cancel out?
# Also can't be definitively sure if genotype is e1/e3 or e2/e4
# n = 11,493
apoe <- apoe[apoe$apoe1==0 & !is.na(apoe$apoe1),]

# Exclude individuals with genetic/reported sex mismatch
# n=337
apoe$Sex <- factor(apoe$Sex, ordered=FALSE)
apoe <- apoe[apoe$Sex == apoe$gender,]



## Other important exclusions

# Exclude individuals with missing BP data
# n = 432
apoe <- apoe[!is.na(apoe$SBP) & !is.na(apoe$DBP),]

# Restrict to self-report "white"
apoe$cauc <- apoe$group=="White"
apoe <- apoe[apoe$cauc==TRUE,]


# Exclude individuals with prevalent dementia
# n = 155
apoe <- apoe[!apoe$VIdementia,]
apoe <- apoe[is.na(apoe$dement_date) | (apoe$dement_date>apoe$recdate & !is.na(apoe$dement_date)),]




# Exclude individuals with no age data
# n = 0
apoe <- apoe[!is.na(apoe$age),]
# Exclude those outside the 40-70 age range
# n = 2247
apoe <- apoe[apoe$age >= 40 & apoe$age < 70,]
# Exclude those who didn't answer smoking or alcohol status
apoe <- apoe[!is.na(apoe$Smo_Status) & apoe$Smo_Status!="Prefer not to answer" ,]
apoe$Smo_Status <- factor(apoe$Smo_Status)
# apoe <- apoe[!is.na(apoe$Alc_Status) & apoe$Alc_Status!="Prefer not to answer" ,]
# apoe$Alc_Status <- factor(apoe$Alc_Status)
apoe <- apoe[!is.na(apoe$Alc_Freq) & apoe$Alc_Freq!="Prefer not to answer" ,]
apoe$Alc_Freq <- factor(apoe$Alc_Freq)
# apoe <- apoe[!is.na(apoe$Sle_Duration) & apoe$Sle_Duration!="Prefer not to answer" ,]
# apoe$Sle_Duration <- factor(apoe$Sle_Duration)
apoe <- apoe[!is.na(apoe$HMH_Diabetes) & apoe$HMH_Diabetes!="Prefer not to answer" & apoe$HMH_Diabetes!="Do not know" ,]
apoe$HMH_Diabetes <- factor(apoe$HMH_Diabetes)
# apoe <- apoe[!is.na(apoe$income) & apoe$income!="Prefer not to answer" & apoe$income!="Do not know" ,]
# apoe$income <- factor(apoe$income)
apoe <- apoe[!is.na(apoe$BSM_BMI) ,]
apoe <- apoe[!is.na(apoe$townsend_depind) ,]

# Remaining n = 474,343
saveRDS(apoe, file="K:\\TEU\\APOE on Dementia\\Data Management\\R_Dataframes_TLA\\38358\\Organised\\apoe_excl.rds")

```

## Intro 
#### Currently stolen from analysis plan May2019

The apolipoprotein E (APOE) e4 allele has been associated with an increased risk of late-onset Alzheimerâ€™s disease (AD) as well as cardiovascular disease (CVD), as it is involved with cholesterol transport and metabolism. Previous studies have reported conflicting results of interaction between APOE e4 and cardiovascular conditions. Identifying interactions between APOE e4 and modifiable CVD risk factors, such as hypertension, on determining cognitive decline would be of considerable health impact, as cognitive decline is one of the markers of clinical dementia. With almost 500,000 participants in the UK Biobank, we wish to examine the interaction between APOE genotypes and hypertension on cognitive function

#### Combine these tabulations into one demographic characteristics table, show mean (SD) for continuous and freq (%) for categorical.

```{r table, include=TRUE, echo=FALSE}
# Tabulate baseline chars
# Columns for number of APOE e4 alles (0, 1, 2)
# Rows for hypertension, age, gender, and all other proposed covariates

kable(aggregate(apoe$age, list(apoe$apoe4), mean, na.rm=TRUE), caption="Mean age vs APOE e4 alleles")
kable(table(apoe$gender, apoe$apoe4, useNA=c('ifany')), caption="Gender vs APOE e4 alleles")
kable(table(apoe$hypertension, apoe$apoe4, useNA=c('ifany')), caption="Hypertension vs APOE e4 alleles")
kable(aggregate(apoe$townsend_depind, list(apoe$apoe4), mean, na.rm=TRUE), caption="Mean Townsend deprivation index vs APOE e4 alleles")
kable(aggregate(apoe$BSM_BMI, list(apoe$apoe4), mean, na.rm=TRUE), caption="Mean BMI vs APOE e4 alleles")
kable(table(apoe$Smo_Status, apoe$apoe4, useNA=c('ifany')), caption="Smoking status vs APOE e4 alleles")
kable(table(apoe$alcdaily, apoe$apoe4, useNA=c('ifany')), caption="Alcohol daily or almost daily vs APOE e4 alleles")
kable(table(apoe$uni, apoe$apoe4, useNA=c('ifany')), caption="Attended university vs APOE e4 alleles")
kable(table(apoe$diab, apoe$apoe4, useNA=c('ifany')), caption="Diabetes vs APOE e4 alleles")
kable(table(apoe$CVD, apoe$apoe4, useNA=c('ifany')), caption="Cardiovascular disease vs APOE e4 alleles")

```

#### Eligibility flowchart?

```{r flowchart, include=FALSE, echo=TRUE}
# Eligibility flowchart?
```

In the Kaplan-Meier curves we see that individuals with hypertension have a higher dementia rate than those without. 

There appears to be a dose-response relationship between dementia and APOE e4 alleles.

```{r km, include=TRUE, echo=FALSE}
# Survival analysis: time to dementia

surv <- Surv(apoe$stime, apoe$status)

ggsurvplot(survfit(surv~apoe4, data=apoe), data=apoe, ylim=c(0.95,1), risk.table=TRUE, ylab="Proportion without dementia", title="Kaplan-Meier plot of time to dementia diagnosis \nby number of APOE e4 alleles", xlab="Time, years", censor=FALSE)

ggsurvplot(survfit(surv~hypertension, data=apoe), data=apoe, ylim=c(0.95,1), risk.table=TRUE, ylab="Proportion without dementia", title="Kaplan-Meier plot of time to dementia diagnosis \nby hypertension status", xlab="Time, years", censor=FALSE)
```

## Methods
We used a Cox regression model to investigate the time until a diagnosis of dementia. Individuals were censored if they were lost to follow-up or died before a dementia diagnosis. Administrative censoring is applied according to the current censoring dates of hospital inpatient data provided on the UKB website.

We considered only individuals who reported their ethnicity as "White", and excluded individuals with prevalent dementia and those with incomplete APOE data. We also excluded individuals who did not answer questions on our key covariates, such as smoking, alcohol frequency and medical conditions. This left us with a sample size of 442,403, of whom 1626 individuals were diagnosed with dementia during the follow-up period.

In the base model we adjust for age and gender. For the fully adjusted model, backwards selection using the likelihood ratio test was used to determine which of our potential covariates improved model fit.

We explore the possibility of an interaction between the number of APOE e4 alleles (an additive model) and hypertension as baseline (defined as any evidence of hypertension, including self-report of a diagnosis, taking hypertensive medications or having hypertensive blood pressure levels at baseline assessment).




```{r basemodel, include=TRUE, echo=FALSE}
coxph(Surv(stime, status) ~ age + gender + hypertension + apoe4, data=apoe)
# Interaction between hypertension and APOE e4
coxph(Surv(stime, status) ~ age + gender + hypertension * apoe4, data=apoe)
```

Incorporating an interaction between hypertensive status and APOE e4 carrier status on the whole data set, we see that although the variables are strongly significant individually, their interaction is not statistically significant and does not substantially improve model fit.


```{r backselectfn, include=FALSE}
# All variables of interest:
# age, sex, assessment centre, education, smoking, depression, alcohol, household income, length of follow-up, pre-existing cardiovascular conditions,
# type 2 diabetes, deprivation scores
covarstring <- function(basemodel, covarlist) {
  if (length(covarlist)>=1) {
    return(paste(basemodel, paste("+", covarlist, collapse=" ")))
  } 
  else {
    return(basemodel)
  }
}

backwardstep <- function(data, basemodel, covarlist) {
  allcovar <- covarstring(basemodel, covarlist)
  all <- coxph(formula(allcovar), data=data)
  pmax <- 0.05
  newlist <- covarlist
  for(covar in covarlist){
    print(covar)
    if(length(covarlist>1)){
      testlist <- covarlist[covarlist!=covar]
      testvar <- covarstring(basemodel, testlist)
    } 
    else {
       testvar <- basemodel
    }
    test <- coxph(formula(testvar), data=data)
    lr <- lrtest(test, all)
    p <- lr$`Pr(>Chisq)`[2]
    print(p)
    if(p > pmax){
      pmax <- p
      newlist <- testlist
    }
  }
  if(length(newlist)<length(covarlist) & length(newlist)>0){
    backwardstep(data=data, basemodel=basemodel, covarlist=newlist)
  }
  else {
    return(newlist)
  }
}

surv_obj <- Surv(apoe$stime, apoe$status)
covarlist <- c("townsend_depind", "BSM_BMI", "Smo_Status", "alcdaily", "uni", "diab", "CVD")

chosencovar <- backwardstep(data=apoe, basemodel="surv_obj ~ age + gender + apoe4 ", covarlist=covarlist)

```

```{r fullmodel, include=TRUE, echo=FALSE}
# No interaction
coxph(formula(covarstring("surv_obj ~ age + gender + hypertension + apoe4 ", chosencovar)), data=apoe)
# Interaction
coxph(formula(covarstring("surv_obj ~ age + gender + hypertension * e4carrier ", chosencovar)), data=apoe)

```

After backward selection, all our considered covariates were found to be significant and to improve model fit. In addition to age, gender, hypertension status and APOE e4 allele count, the resulting adjusted model therefore included Townsend deprivation index, BMI, smoking status, alcohol frequency, whether the participant had attended university, and whether they had self-reported diabetes or cardiovascular disease.

Once again, the interaction between APOE e4 alleles and hypertensive status was not statistically significant.

The individual hazard ratios for APOE e4 alleles are not materially altered by adjusting for other covariates, but the hazard ratio for hypertension was attenuated in the fully adjusted model, reducing from around 1.4 to around 1.2.

Inspection of the Schoenfeld residuals did not indicate any violation of the proportional hazards assumption.

```{r residuals, include=FALSE, echo=FALSE}
cox <- coxph(formula(covarstring("surv_obj ~ age + gender + hypertension + apoe4 ", chosencovar)), data=apoe)
test.ph <- cox.zph(cox)
ggcoxzph(test.ph)
res <- residuals(cox, type="schoenfeld")
```


```{r incidencerates, include=TRUE, echo=FALSE}
table(apoe$hypertension[apoe$status==TRUE], apoe$apoe4[apoe$status==TRUE])
table(apoe$hypertension, apoe$apoe4)

table(apoe$hypertension[apoe$status==TRUE], apoe$e4carrier[apoe$status==TRUE])
table(apoe$hypertension, apoe$e4carrier)

apoe$joint <- interaction(as.numeric(apoe$e4carrier), as.numeric(apoe$hypertension))
apoe$joint <- factor(as.numeric(apoe$joint)-1, labels=c("not a carrier, no hypertension", "e4 carrier, no hypertension", "not a carrier, hypertensive", "e4 carrier, hypertensive"))
coxph(Surv(stime, status) ~ age + gender + joint, data=apoe)

add <- aalen(Surv(stime, status) ~ age + gender + apoe4 + hypertension, data=apoe)
summary(add)
plot(add)

```

## Notes

- Currently using BMI as a continuous variable, but maybe should dichotomise clinically obese/not?
- Could add sleep duration as a potential covariate? Would need categorising - too little, normal, too much?
- Why does daily drinking of alcohol appear to have protective effect?

## Further work

- Adjust for stratification with genetic principal components?
- Consider physical activity as a confounder?
- Consider the measures of cognitive function - fluid intelligence etc
- Consider alzheimers and vascular dementia separately?
- Check for influential observations then sensitivity analysis?
- Higher order term for age? 
- Consider modelling with splines?
- Genetic instead of self-reported ethnicity

# References
